app/helpers/application_helper.rb

```diff
@@ -1,2 +1,7 @@
 module ApplicationHelper
+  def flash_message
+    flash.map do |key, message|
+      content_tag :p, message, :class => key
+    end.join().html_safe
```

Comment:
Thanks for advice in points 1-3.
Regarding 4., I believe that's what @gotar meant [here](https://github.com/Ragnarson-Internship2016/group-I/pull/41#discussion_r71380181).